### ISSUE TITLE
[BUG]The larger batch size, the slower training speed?

### ISSUE THREAD

**Issue by Hap-Zhang** [microsoft/DeepSpeed#4036]
I trained a llama 7B model with deepspeed zero3 in A100 80G * 8, and training script  is https://github.com/tatsu-lab/stanford_alpaca/blob/main/train.py

I set batch size 8 and gradient checkpoint is true, the throughout is 64.101samples/sec.
![image](https://github.com/microsoft/DeepSpeed/assets/32810063/be0ab8e2-9654-4428-b2bd-39ca317d15d0)

However, when i set batch size 64 and gradient checkpoint is true, the throughout drops to 46.86 samples/sec.
![image](https://github.com/microsoft/DeepSpeed/assets/32810063/e209bfc7-118f-4969-956f-837f3148ec39)

Is this result as expected?  I think the larger the batch size will get  faster training speed with same number of samples.

**Comment by Modas-Li** [comment]
Does your llama 7B model support chinese task after fine-tuning? If so, then how? Thx~

**Comment by Hap-Zhang** [comment]
@Modas-Li Sorry, i just compare the performance of different training frameworks. So,  i didn't pay attention to chinese task.

**Comment by jomayeri** [comment]
@Hap-Zhang Check the memory usage. If the batch size gets too large and consumes too much memory it triggers the python garbage collector and slows down runtime.
